### PR TITLE
feat: refactor JavaScript implementation and add C `ndarray` implementation for `blas/base/sasum`

### PR DESCRIPTION
**Pull request by aman-095**
Progresses #2039.

## Description

> What is the purpose of this pull request?

This pull request:

-   updates existing Js implementation based on the latest lapack implementation.
-   add C `ndarray` API.
-   update documentation.

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   progresses #2039  

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/blas/base/sasum/lib/ndarray.js [hunk 1]
```diff
@@ -47,14 +52,36 @@ function sasum( N, x, stride, offset ) {
...
 	var i;
+	var m;
 
 	sum = 0.0;
 	if ( N <= 0 ) {
 		return sum;
 	}
 	ix = offset;
+
+	// If the stride is equal to `1`, use unrolled loops...
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/base/sasum/lib/ndarray.js]: @aman-095 You should revert this change. Loop unrolling makes the most sense for simple arithmetic operations. Here, L78 is incorrect and requires `f32()` wrapping for each `+` operation. Better to just use a simple `for` as at L83.

File: lib/node_modules/@stdlib/blas/base/sasum/test/test.sasum.js [hunk 2]
```diff
@@ -82,15 +82,23 @@ tape( 'if provided an `N` parameter less than or equal to `0`, the function retu
...
+tape( 'the function supports specifying a negative stride', function test( t ) {
 	var x;
 	var y;
+	var N;
 
-	x = new Float32Array( [ 1.0, -2.0, 3.0, -4.0, 5.0 ] );
+	x = new Float32Array([
+		1.0,  // 2
+		-2.0,
+		3.0,  // 1
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/base/sasum/test/test.sasum.js]:
```suggestion
		1.0,  // 1
		-2.0,
		3.0,  // 0
```

File: lib/node_modules/@stdlib/blas/base/sasum/test/test.sasum.native.js [hunk 3]
```diff
@@ -91,15 +91,23 @@ tape( 'if provided an `N` parameter less than or equal to `0`, the function retu
...
+tape( 'the function supports specifying a negative stride', opts, function test( t ) {
 	var x;
 	var y;
+	var N;
 
-	x = new Float32Array( [ 1.0, -2.0, 3.0, -4.0, 5.0 ] );
+	x = new Float32Array([
+		1.0,  // 2
+		-2.0,
+		3.0,  // 1
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/base/sasum/test/test.sasum.native.js]:
```suggestion
		1.0,  // 1
		-2.0,
		3.0,  // 0
```

**Review by kgryte — CHANGES_REQUESTED** [review]
Apart from the JS ndarray implementation, this PR looks good to go. Once the implementation is updated, should be ready for merge.

File: lib/node_modules/@stdlib/blas/base/sasum/lib/ndarray.js (hunk 1 above)
Comment (aman-095) [reply lib/node_modules/@stdlib/blas/base/sasum/lib/ndarray.js]: Should I also revert the changes for `sasum_ndarray.c` or is it fine.

Comment (kgryte) [reply lib/node_modules/@stdlib/blas/base/sasum/lib/ndarray.js]: The C implementation is fine, as we are using single-precision arithmetic and don't have the extraneous casting calls.

**Review by kgryte — APPROVED** [review]
LGTM. Thanks, @aman-095!
